Statsman/PeriodStatisticsService.cs

```diff
@@ -0,0 +1,140 @@
+        }
+
+        public async Task<(Stream Chart, ResultMessage Message)> PerHour(StatisticsRequest request)
+        {
+            var timeRange = TimeRange.Create(DateTime.UtcNow.AddHours(-request.TimeBehind.TotalHours), DateTime.UtcNow);
+            var statistics = await this.GetStatisticsGroupedPerDetailedPeriod(request, timeRange, Period.Hour);
+            return (Chart: await this._chartsService.GetImageStatisticsPerPeriod(statistics, "Messages per hour"),
+                Message: this.GetMessage(request.UserId, request.ChannelId, Period.Hour, timeRange));
+        }
+
+        public async Task<(Stream Chart, ResultMessage Message)> PerDay(StatisticsRequest request)
+        {
+            var timeRange = TimeRange.Create(DateTime.Today.AddDays(-request.TimeBehind.TotalDays), DateTime.UtcNow);
+            var statistics = await this.GetStatisticsGroupedPerDaysPeriod(request, timeRange, Period.Day);
+            return (Chart: await this._chartsService.GetImageStatisticsPerPeriod(statistics, "Messages per day"),
+                Message: this.GetMessage(request.UserId, request.ChannelId, Period.Day, timeRange));
+        }
+
+        public async Task<(Stream Chart, ResultMessage Message)> PerWeek(StatisticsRequest request)
+        {
+            var timeRange = TimeRange.Create(DateTime.Today.AddDays(-request.TimeBehind.TotalDays), DateTime.UtcNow);
+            var statistics = await this.GetStatisticsGroupedPerDaysPeriod(request, timeRange, Period.Week);
+            return (Chart: await this._chartsService.GetImageStatisticsPerPeriod(statistics, "Messages per week"),
+                Message: this.GetMessage(request.UserId, request.ChannelId, Period.Week, timeRange));
+        }
+
+        public async Task<(Stream Chart, ResultMessage Message)> PerMonth(StatisticsRequest request)
+        {
+            var timeRange = TimeRange.Create(DateTime.Today.AddDays(-request.TimeBehind.TotalDays), DateTime.UtcNow);
+            var statistics = await this.GetStatisticsGroupedPerDaysPeriod(request, timeRange, Period.Month);
+            return (Chart: await this._chartsService.GetImageStatisticsPerPeriod(statistics, "Messages per month"),
+                Message: this.GetMessage(request.UserId, request.ChannelId, Period.Month, timeRange));
+        }
+
+        public async Task<(Stream Chart, ResultMessage Message)> PerQuarter(StatisticsRequest request)
+        {
+            var timeRange = TimeRange.Create(DateTime.Today.AddDays(-request.TimeBehind.TotalDays), DateTime.UtcNow);
+            var statistics = await this.GetStatisticsGroupedPerDaysPeriod(request, timeRange, Period.Quarter);
+            return (Chart: await this._chartsService.GetImageStatisticsPerPeriod(statistics, "Messages per quarter"), 
+                Message: this.GetMessage(request.UserId, request.ChannelId, Period.Quarter, timeRange));
+        }
```

Comment:
w tym miejscu można wydzielić do innej metody wspólny kod. W zasadzie tutaj dzieje się to samo tylko na innych danych.